From 7862d1b1487138a97d4f1b3e8d6012230e5e8307 Mon Sep 17 00:00:00 2001
From: Liu Ying <Ying.Liu@freescale.com>
Date: Mon, 30 Nov 2015 10:18:58 +0800
Subject: [PATCH] MLK-11918-1 video: fbdev: mxc ipuv3 fb: Remove fbi NULL
 pointer check in ->remove()

The driver takes the pointer fbi as the driver data, so fbi cannot be a NULL
pointer in ->remove().  Let's remove the unnecessary NULL pointer check on
fbi.

This patch fixes the following issue reported by Coverity:
deref_ptr: Directly dereferencing pointer fbi.
	struct mxcfb_info *mxc_fbi = fbi->par;

Dereference before null check (REVERSE_INULL)
check_after_deref: Null-checking fbi suggests that it may be null, but it has
already been dereferenced on all paths leading to the check.
	if (!fbi)
		return 0;

Signed-off-by: Liu Ying <Ying.Liu@freescale.com>
(cherry picked from commit 5a05f2e6b4a031544e7c795ea98f558fc2a3666d)
---
 drivers/video/mxc/mxc_ipuv3_fb.c | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/drivers/video/mxc/mxc_ipuv3_fb.c b/drivers/video/mxc/mxc_ipuv3_fb.c
index eb9b2f2..9f71693 100644
--- a/drivers/video/mxc/mxc_ipuv3_fb.c
+++ b/drivers/video/mxc/mxc_ipuv3_fb.c
@@ -3591,9 +3591,6 @@ static int mxcfb_remove(struct platform_device *pdev)
 	struct fb_info *fbi = platform_get_drvdata(pdev);
 	struct mxcfb_info *mxc_fbi = fbi->par;
 
-	if (!fbi)
-		return 0;
-
 	device_remove_file(fbi->dev, &dev_attr_fsl_disp_dev_property);
 	device_remove_file(fbi->dev, &dev_attr_fsl_disp_property);
 	mxcfb_blank(FB_BLANK_POWERDOWN, fbi);
-- 
1.8.0

